src/components/CustomCheckout.tsx

```diff
@@ -0,0 +1,275 @@
+   * Once this prop has been set, it can not be changed.
+   *
+   * You can also pass in `null` or a `Promise` resolving to `null` if you are performing an initial server-side render or when generating a static site.
+   */
+  stripe: PromiseLike<stripeJs.Stripe | null> | stripeJs.Stripe | null;
+  options: stripeJs.StripeCustomCheckoutOptions;
+}
+
+interface PrivateCustomCheckoutProviderProps {
+  stripe: unknown;
+  options: stripeJs.StripeCustomCheckoutOptions;
+  children?: ReactNode;
+}
+const INVALID_STRIPE_ERROR =
+  'Invalid prop `stripe` supplied to `CustomCheckoutProvider`. We recommend using the `loadStripe` utility from `@stripe/stripe-js`. See https://stripe.com/docs/stripe-js/react#elements-props-stripe for details.';
+
+export const CustomCheckoutProvider: FunctionComponent<PropsWithChildren<
+  CustomCheckoutProviderProps
+>> = (({
+  stripe: rawStripeProp,
+  options,
+  children,
+}: PrivateCustomCheckoutProviderProps) => {
+  const parsed = React.useMemo(
+    () => parseStripeProp(rawStripeProp, INVALID_STRIPE_ERROR),
+    [rawStripeProp]
+  );
+
+  // State used to trigger a re-render when sdk.session is updated
+  const [
+    _,
+    setSession,
+  ] = React.useState<stripeJs.StripeCustomCheckoutSession | null>(null);
+
+  // State used to avoid calling initCustomCheckout multiple times when options changes
+  const [
+    initCustomCheckoutCalled,
+    setInitCustomCheckoutCalled,
+  ] = React.useState<boolean>(false);
+
+  const [ctx, setContext] = React.useState<CustomCheckoutSdkContextValue>(
+    () => ({
+      stripe: parsed.tag === 'sync' ? parsed.stripe : null,
+      customCheckoutSdk: null,
+    })
+  );
+
+  const safeSetContext = (
+    stripe: stripeJs.Stripe,
+    customCheckoutSdk: stripeJs.StripeCustomCheckout
+  ) => {
+    setContext((ctx) => {
+      if (ctx.stripe && ctx.customCheckoutSdk) {
+        return ctx;
+      }
+
+      return {stripe, customCheckoutSdk};
+    });
+  };
+
+  React.useEffect(() => {
+    let isMounted = true;
+
+    if (parsed.tag === 'async' && !ctx.stripe) {
+      parsed.stripePromise.then((stripe) => {
+        if (stripe && isMounted && !initCustomCheckoutCalled) {
+          // Only update context if the component is still mounted
+          // and stripe is not null. We allow stripe to be null to make
+          // handling SSR easier.
+          setInitCustomCheckoutCalled(true);
+          stripe.initCustomCheckout(options).then((customCheckoutSdk) => {
+            if (customCheckoutSdk) {
+              safeSetContext(stripe, customCheckoutSdk);
+              customCheckoutSdk.on('change', setSession);
+            }
+          });
+        }
+      });
+    } else if (
+      parsed.tag === 'sync' &&
+      parsed.stripe &&
+      !initCustomCheckoutCalled
+    ) {
+      setInitCustomCheckoutCalled(true);
+      parsed.stripe.initCustomCheckout(options).then((customCheckoutSdk) => {
+        if (customCheckoutSdk) {
```

Comment:
```suggestion
        if (customCheckoutSdk && isMounted) {
```

Reply:
Similar reasons here, because we call `setInitCustomCheckoutCalled` here https://github.com/stripe/react-stripe-js/blob/26bb4e222db70e47026f2fb03f99c72888374013/src/components/CustomCheckout.tsx#L143-L145

So if we check isMounted here, it's false when initCustomCheckout is resolved and it will NOT call safeSetContext